### PR TITLE
Enable rubygems_mfa_required on solidus

### PR DESCRIPTION
**Pull request by gsmendoza**
Goal
----

As a Solidus maintainer
I want to enable rubygems_mfa_required on the gem's gemspec
So that MFA will be required when performing privileged operations on the gem

Reference
---------

https://docs.rubocop.org/rubocop/cops_gemspec.html#gemspecrequiremfa

https://guides.rubygems.org/mfa-requirement-opt-in/

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- N/A: I have updated Guides and README accordingly to this change (if needed)
- N/A: I have added tests to cover this change (if needed)
- N/A: I have attached screenshots to this PR for visual changes (if needed)

### CODE REVIEW

File: solidus.gemspec [hunk 1]
```diff
@@ -14,6 +14,8 @@ Gem::Specification.new do |s|
   s.homepage    = 'http://solidus.io'
   s.license     = 'BSD-3-Clause'
 
+  s.metadata['rubygems_mfa_required'] = 'true'
```
Comment (kennyadsl) [inline solidus.gemspec:17]: I think this is needed in all the other subgems as well, isn't it?

**Review by kennyadsl — CHANGES_REQUESTED** [review]

File: solidus.gemspec (hunk 1 above)
Comment (gsmendoza) [reply solidus.gemspec:17]: Oh, okay, I'll update the subgems too :+1: TODO

**Review by jarednorman — APPROVED** [review]
Thanks!

**Review by waiting-for-dev — APPROVED** [review]
Thanks! :heart:

**Review by kennyadsl — APPROVED** [review]
👍

**Pull request merged**